scenarios/kubernetes_e2e.py

```diff
@@ -309,14 +329,81 @@ def sig_handler(self, _signo, _frame):
         check('docker', 'stop', self.container)
 
 
-def cluster_name(cluster, build):
+def cluster_name(cluster, build, domain=None):
     """Return or select a cluster name."""
     if cluster:
-        return cluster
-    if len(build) < 20:
-        return 'e2e-%s' % build
-    return 'e2e-%s' % hashlib.md5(build).hexdigest()[:10]
+        clu = cluster
+    elif len(build) < 20:
+        clu = 'e2e-%s' % build
+    else:
+        clu = 'e2e-%s' % hashlib.md5(build).hexdigest()[:10]
+    if domain:
+        clu = '%s.%s' % (clu, domain)
+    return clu
+
+def get_aws_zone(zone):
+    if zone:
+        return zone
+    return random.choice([
+        'us-west-1a',
+        'us-west-1c',
+        'us-west-2a',
+        'us-west-2b',
+        'us-east-1a',
+        'us-east-1d',
+        #'us-east-2a',
+        #'us-east-2b',
+    ])
+
+def build_kops(workspace, mode):
+    """Build kops, set kops related envs."""
+    if not os.path.basename(workspace) == 'kops':
+        raise ValueError(workspace)
+    version = 'pull-' + check_output('git', 'describe', '--always').strip()
+    job = os.getenv('JOB_NAME', 'pull-kops-e2e-kubernetes-aws')
+    gcs = 'gs://kops-ci/pulls/%s' % job
+    gapi = 'https://storage.googleapis.com/kops-ci/pulls/%s' % job
+    mode.add_environment([
+        'KOPS_BASE_URL=%s/%s' % (gapi, version),
+        'GCS_LOCATION=%s' % gcs
+        ])
+    check('make', 'gcs-publish-ci', 'VERSION=%s' % version, 'GCS_LOCATION=%s' % gcs)
+
+def set_up_aws(args, mode, cluster, runner_args):
+    """Set up aws related envs."""
+    for path in [args.aws_ssh, args.aws_pub, args.aws_cred]:
+        if not os.path.isfile(os.path.expandvars(path)):
+            raise IOError(path, os.path.expandvars(path))
+    mode.add_aws_cred(args.aws_ssh, args.aws_pub, args.aws_cred)
+
+    aws_config = '/workspace/.aws/config'
+    aws_ssh = '/workspace/.ssh/kube_aws_rsa'
+    profile = args.aws_profile
+    if args.aws_role_arn:
+        with tempfile.NamedTemporaryFile(
+            prefix='aws-config', delete=False) as cfg:
+            cfg.write(AWS_ROLE_CONFIG.format(
+                role_arn=args.aws_role_arn,
+                source_profile=profile))
+            mode.add_aws_profile(cfg.name, aws_config)
+    profile = 'jenkins-assumed-role' # From AWS_ROLE_CONFIG
+
+    zones = get_aws_zone(args.aws_zones)
+    regions = ','.join([zone[:-1] for zone in zones.split(',')])
 
+    mode.add_environment(
+      'AWS_PROFILE=%s' % profile,
+      'AWS_DEFAULT_PROFILE=%s' % profile,
+      'KOPS_REGIONS=%s' % regions,
+    )
+
+    runner_args.extend([
+        '--kops-cluster=%s' % cluster,
+        '--kops-zones=%s ' % zones,
```

Comment:
Why do we call it --aws-zones at the scenario level and --kops-zones at the kubetest level? Can we call them both --kops-zones? Same for the other --aws flags?